js/GameObject.js

```diff
@@ -0,0 +1,15 @@
+class GameObject {
+  constructor(parent, width, height, position = new Vec2(), image = new Image(), velocity = new Vec2(),
+   acceleration = new Vec2(), angularRotation = 0, angularVelocity = 0, angularAcceleration = 0){

```

Comment:
change angularRotation to angularPosition. Also Angular Position is more important than Velocity and accelaration. Also add axis of rotation relative to object Position